.gitignore

```diff
@@ -1,3 +1,4 @@
 .DS_Store
-.idea/workspace.xml
```

Comment:
which command is issued to generate the AutoGeneratedDomainInfor.java (and the other two files) ?

Reply:
Pantheon gitignores any IDE-specific directory (IDEA, Eclipse, etc) and autogenerated ones, including  .gradle . 

IntelliJ support gives half-baked tips about what to ignore or not, noting that gradle projects are autoimported anyway, and getting lots of discussion: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems 

I'd vote to follow the Pantheon way for simplicity. Probably tool configuration should be more global anyway, if we are going to share it.

Reply:
Yes, in Pantheon, gradle is the source of truth and IntelliJ imports gradle configurations. Similar logic applies to the the "src" directory here which is truffle driven.

So I vote for removing .idea/ directories from git for projects that are driven by CLI's like gradle and truffle.

This would mean removing wrappers/java/.idea directory as well. Just tested it and I was able to build the java wrapper without any problems (IntelliJ automatically offers to import it).

Reply:
We all decided to keep IDE/autogenerated files out of the repo.